Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py

```diff
@@ -75,84 +75,103 @@ def __init__(self, url: str = '', credentials: dict = None,
+            **kwargs
+        )
+
+        try:
+            r.raise_for_status()
+            data = r.json()
+            result = jmespath.search(expression=feed.get('extractor'), data=data)
+
+        except ValueError as VE:
+            raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {VE}')
+
+        return result
+
+
+def test_module(client: Client, limit) -> str:
+    for feed_name, feed in client.feed_name_to_config.items():
+        build_iterator_paging = feed.get('build_iterator_paging')
+        if build_iterator_paging:
+            build_iterator_paging(client, feed, limit)
+        else:
+            client.build_iterator(feed)
     return 'ok'
 
 
-def fetch_indicators_command(client: Client, indicator_type: str, feedTags: list, auto_detect: bool, **kwargs) \
-        -> Union[Dict, List[Dict]]:
+def fetch_indicators_command(client: Client, indicator_type: str, feedTags: list, auto_detect: bool,
+                             limit: int = 0, **kwargs) -> Union[Dict, List[Dict]]:
     """
     Fetches the indicators from client.
     :param client: Client of a JSON Feed
     :param indicator_type: the default indicator type
     :param feedTags: the indicator tags
     """
     indicators = []
-    for result in client.build_iterator(**kwargs):
-        for service_name, items in result.items():
-            feed_config = client.feed_name_to_config.get(service_name, {})
-            indicator_field = feed_config.get('indicator') if feed_config.get('indicator') else 'indicator'
-            indicator_type = feed_config.get('indicator_type', indicator_type)
-            for item in items:
-                mapping = feed_config.get('mapping')
-
-                if isinstance(item, str):
-                    item = {indicator_field: item}
-                indicator_value = item.get(indicator_field)
-
-                current_indicator_type = determine_indicator_type(indicator_type, auto_detect, indicator_value)
-                if not current_indicator_type:
-                    continue
-
-                indicator = {
-                    'value': indicator_value,
-                    'type': current_indicator_type,
-                    'fields': {
-                        'tags': feedTags,
-                    }
+    feeds_results = {}
+    for feed_name, feed in client.feed_name_to_config.items():
+        build_iterator_paging = feed.get('build_iterator_paging')
+        if build_iterator_paging:
+            demisto.debug("Got a custom function to handle with pagination, "
+                          "Going to use this function instead build_iterator function")
+            indicators_from_feed = build_iterator_paging(client, feed, limit, **kwargs)
+            if isinstance(indicators_from_feed, list):
+                feeds_results[feed_name] = indicators_from_feed
+            else:
+                demisto.debug("Custom function to handle with pagination must have a list as a return type")
```

Comment:
not sure this is needed, if a dev implemented it wrongfully, the linter will shout at him - why do we need this check in the code we release?